### DISCUSSION TITLE
LF APAC 开源布道者分享系列之十一：上游思维与上游优先 - 开源之道

### DISCUSSION THREAD

**Discussion by giscus** [OCselected/ocselected.github.io#16] [Announcements]
# LF APAC 开源布道者分享系列之十一：上游思维与上游优先 - 开源之道

https://opensourceway.community/posts/lfapac-evangelist-sharing/upstream-first-and-be-upstream-in-os/

**Comment by ZoomQuiet** [comment]
#是也乎,(￣▽￣)
其实, UpStream 翻译为"逆流" 可能更加好些;
因为:
开源项目作为直接可以用的代码来源可以说是上游,
但是, 并不是所有开源项目都是底层基础项目, 很多现代项目直接就是产品级的,
是在其它更多开源上游项目基础上的最终应用, 在技术体系中, 其实是下游...

文章提出的几个主张:
+ 思想要意识到有上游
+ 行为要注意及时和上游合并
+ 策略上得自觉成为上游...

其实, 相同的状态在:

- 随时有逆流而动的决心, 不惜代价从难作起, 不受制与历史, 自在逍遥...
- UpStream 这词儿实话包含同步上下交流的含义, 这不是中文 上游 这个名词可以包含的

上游思想, 其实就是常说的:

+ 想事儿, 要高一个层级
+ 作事儿, 低高一个层级

无时不刻, 对世界保持敬意, 下决心时,心怀广袤些, 开始作事儿时又值得认真从源头开始, 
顺着前人冲开的正确方向自然协同, 别老想着 "自主XXX"

**Comment by ZoomQuiet** [comment]
|> 231216 OSCAR 微信群中一个截屏引发了争论:

> 参与开源社区项目 : 请记住，根据我们的《准则》,参与开源社区项目,无论项目是由我们公司还是由其他商业或非商业实体或组织开展的，都不构成利益冲突，您甚至可以做出有利于项目但不利于公司利益的决定。

- 正方: 上游优先是开源正义, 是绝对的不容质疑
- 反方: 在企业场景中, 商业利益为先, 这是唯一底线
- 中方: 上游优先是开源现象, 在不同场景中是可以不同应对的

举行个例子, 为什么 Apache Log4j2 史诗级漏洞至今没有好办法解决?
明明是个上游关键组件, 为什么多年无人关注, 却有那么多项目在使用?

上游只是在心理/地理/历史/... 上的优先出没,
并不是绝对真理;

再比如 Android 是真正的上游吧,
可 google 就用一层很薄的专有协议层, 轻松完成了隔离, 并建立联盟,
有效布局对抗 iOS 生态,
可依然无法扼制新生态的发韧;

上游为先, 当然好;

不过, 那得在所有条件都吻合的情况下,
否则, 太多姿势可以绕过/污化/超级/替代了....

**Reply by lijiangsheng1** [reply]
一种思维方式，也是一种永远值得追求的境界。
